discovery(backend): /api/health api_route uses single operation_id for GET+HEAD — causes duplicate identifier in generated OpenAPI types

## Problem

Found while implementing #5209 (OpenAPI-driven frontend type generation).

The backend registers `/api/health` with both GET and HEAD methods via a single `@app.api_route(...)` decorator:

\`\`\`python
# autobot-backend/initialization/endpoints.py:123
@app.api_route("/api/health", methods=["GET", "HEAD"])
async def root_health_check(request: Request):
    ...
\`\`\`

FastAPI auto-generates the same \`operationId\` for both HTTP methods:

\`\`\`bash
$ curl -s http://127.0.0.1:8001/openapi.json | jq '.paths."/api/health" | {get: .get.operationId, head: .head.operationId}'
{
  "get": "root_health_check_api_health_head",
  "head": "root_health_check_api_health_head"
}
\`\`\`

This violates the OpenAPI 3.1 spec (operationIds must be unique) and causes \`openapi-typescript\` to emit duplicate identifiers in the generated TS file:

\`\`\`
src/types/generated/api.ts(59263,5): error TS2300: Duplicate identifier 'root_health_check_api_health_head'.
src/types/generated/api.ts(59283,5): error TS2300: Duplicate identifier 'root_health_check_api_health_head'.
\`\`\`

## Fix

Split into two route decorators with distinct names, or pass an explicit \`operation_id\` to one of them:

\`\`\`python
@app.get("/api/health", operation_id="root_health_check_get")
async def root_health_check_get(request: Request): ...

@app.head("/api/health", operation_id="root_health_check_head")
async def root_health_check_head(request: Request): ...
\`\`\`

Alternatively, grep the codebase for other multi-method \`@app.api_route\` usages — any route with \`methods=[...]\` containing more than one method is a collision candidate.

## Acceptance criteria

- \`curl /openapi.json | jq '[.paths[] | to_entries[] | .value.operationId] | group_by(.) | map(select(length > 1))' \` returns \`[]\`
- \`npm run gen:types && npx vue-tsc --noEmit\` produces no \`TS2300 Duplicate identifier\` errors on the generated file

## Meta
- Found during: #5209
- Affected files: \`autobot-backend/initialization/endpoints.py:123\`

--- a/autobot-backend/initialization/endpoints.py
+++ b/autobot-backend/initialization/endpoints.py
@@ -120,11 +120,8 @@ def register_root_endpoints(app: FastAPI) -> None:
         app: FastAPI application instance
     """
 
-    @app.api_route("/api/health", methods=["GET", "HEAD"])
-    @with_error_handling(category=ErrorCategory.SYSTEM)
-    async def root_health_check(request: Request):
-        """Health endpoint with per-service breakdown and capabilities (supports GET and HEAD)."""
-        state = request.app.state
+    def _health_payload(state: Any) -> Dict[str, Any]:
+        """Build the /api/health response body (shared by GET and HEAD)."""
         services = _build_services_status(state)
         ai_stack_ready = services.get("ai_stack") == "connected"
         ai_agents_ready = services.get("ai_stack_agents") == "ready"
@@ -138,6 +135,18 @@ def register_root_endpoints(app: FastAPI) -> None:
             "capabilities": _build_capabilities(ai_stack_ready, ai_agents_ready),
             "circuit_breakers": _get_circuit_breaker_states(),
         }
+
+    @app.get("/api/health", operation_id="root_health_check_get")
+    @with_error_handling(category=ErrorCategory.SYSTEM)
+    async def root_health_check_get(request: Request):
+        """Health endpoint with per-service breakdown and capabilities."""
+        return _health_payload(request.app.state)
+
+    @app.head("/api/health", operation_id="root_health_check_head", include_in_schema=False)
+    @with_error_handling(category=ErrorCategory.SYSTEM)
+    async def root_health_check_head(request: Request):
+        """HEAD variant for liveness probes — FastAPI strips the body automatically."""
+        return _health_payload(request.app.state)
 
     @app.get("/api/health/ai-stack")
     @with_error_handling(category=ErrorCategory.SYSTEM)

